### ISSUE TITLE
How to measure bits / dim?

### ISSUE THREAD

**Issue by pclucas14** [nadavbh12/VQ-VAE#7]
Hi, 

first of all, thanks for your code. I've been using it for a little while and it works great. I'm trying to convert the loss to bits / dim in order to get an idea of how the model does w.r.t to the "true" values. Do you know to calculate this ?

Thanks in advance, 
Lucas

**Comment by nadavbh12** [comment]
Hey Lucas,
You're welcome :-) Sorry for the late reply.
To measure this, you need to replace the MSE loss with and NLL loss.
You can this of this as performing classification for every pixel to 256 "classes" (pixels).
Then you can calculate it in the same manner used in [PixelCNN++](https://openreview.net/pdf?id=BJrFC6ceg) (they have tensorflow code [here](https://github.com/openai/pixel-cnn)).

**Comment by pclucas14** [comment]
Hi, 

I guess my question is more "how do you interpret MSE as NLL loss" ? If you assume the outputs are from a gaussian, is there an easy way to get bits / dim? 

thanks for your answer btw!
Lucas

**Comment by nadavbh12** [comment]
This wasn't entirely clear from the paper, so I contacted the authors that verified that for the CIFAR10 experiments, they used cross entropy loss in order to compare to VAE, for imagenet, they used MSE.

**Comment by pclucas14** [comment]
ok awesome!

closing the issue.
